### PR TITLE
feat: add opt-in telemetry, registration, and heartbeat

### LINKED ISSUE #15: Phase 2a: add opt-in telemetry, registration, and heartbeat
## Goal

Implement the opt-in telemetry, registration, and heartbeat flow in the OSS repo for the Phase 2a evidence loop.

## Meta References

- Execution issue: https://github.com/tborys/driftshield-meta/issues/30
- Phase parent: https://github.com/tborys/driftshield-meta/issues/15
- Metrics semantics: https://github.com/tborys/driftshield-meta/issues/32
- Execution map: https://github.com/tborys/driftshield-meta/issues/19

## Acceptance Criteria

- [ ] Telemetry is explicitly opt-in and disabled by default
- [ ] Registration and heartbeat behavior are defined clearly enough for the first evidence-collection loop
- [ ] The initial event or field inventory is documented and intentionally small
- [ ] The implementation is aligned with the Phase 2a metrics model

## Verification

- [ ] A smoke path shows telemetry emits only after opt-in is enabled
- [ ] A smoke path shows no telemetry is sent when opt-in remains disabled
- [ ] The telemetry fields used for Phase 2a are documented or linked

## Out Of Scope

- Broad product analytics or growth tooling
- Non-consensual background collection

## Agent Kickoff

Define the minimum useful event inventory and the consent boundary first. Keep the transport simple and the opt-in path trustworthy.

- Telemetry docs: `driftshield/docs/telemetry.md`

## Completion

Merged in PR #30: https://github.com/tborys/driftshield/pull/30

### PR DESCRIPTION
**Pull request by tborys**
## Summary
- add an explicitly opt-in Phase 2a telemetry service with local registration, heartbeat, and analysis-result emission
- add a `driftshield telemetry` CLI surface for consent status, opt-in, heartbeat, and smoke testing
- document the minimal Phase 2a event inventory and add targeted CLI tests for disabled and enabled behaviour

## Testing
- `PYTHONPATH=src pytest tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

Closes #15

### CODE REVIEW

**Comment by tborys** [comment]
@codex review

**Comment by tborys** [comment]
Fixed the failing public-scope check.

What changed:
- removed private meta-issue and private-doc references from `driftshield/docs/telemetry.md`
- kept the public doc self-contained while preserving the Phase 2a field inventory intent

Verification:
- `./scripts/check-public-scope.sh`

**Comment by tborys** [comment]
@codex review

**Comment by tborys** [comment]
Addressed the valid Codex finding.

Fixed now:
- validate `outcome_status` against the documented Phase 2a enum before emitting `analysis_result` events
- reject invalid CLI input instead of writing polluted telemetry
- add a regression test for invalid status rejection

Rejected as stale:
- the private-doc reference finding was valid on the earlier commit and has already been fixed in `driftshield/docs/telemetry.md`

Verification:
- `PYTHONPATH=src pytest tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

**Comment by tborys** [comment]
@codex review

**Comment by tborys** [comment]
Addressed the remaining valid Codex finding.

Fixed now:
- derive `classifiable` from the validated, normalized `outcome_status`
- add a regression test for whitespace-normalized status input

Verification:
- `PYTHONPATH=src pytest tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

**Comment by tborys** [comment]
@codex review

File: driftshield/docs/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+- `event_type = heartbeat`
+- `occurred_at`
+- `install_id`
+- `payload.status = alive`
+- `payload.event_inventory_version = phase-2a-v1`
+
+### Analysis result
+
+Used as the Phase 2a smoke path for metric-shaped event fields before full instrumentation in `driftshield-meta#33`.
```
[the remark this answers is not in the document]
Comment (tborys) [reply driftshield/docs/telemetry.md]: This is already addressed on `main`. The public telemetry doc no longer references the private planning repo, so the OSS boundary leak called out here has been removed. Resolving as fixed.

File: driftshield/src/driftshield/cli/commands/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    emitted = TelemetryService().heartbeat()
+    if not emitted:
+        console.print("Telemetry is disabled; heartbeat not emitted.")
+        raise typer.Exit(1)
+    console.print("Heartbeat emitted.")
+
+
+@app.command("emit-analysis")
+def telemetry_emit_analysis(
+    outcome_status: str = typer.Option(..., "--outcome-status", help="matched, unclassified, or not_classifiable"),
```
[the remark this answers is not in the document]
Comment (tborys) [reply driftshield/src/driftshield/cli/commands/telemetry.py:66]: This is already addressed on `main`. The CLI now validates `--outcome-status` with `validate_outcome_status(...)` before emitting an event, and the service validates again as a second guard. Resolving as fixed.

File: driftshield/src/driftshield/telemetry/service.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        if not config.enabled or not config.install_id:
+            return False
+        self._write_event(
+            TelemetryEvent(
+                event_type="analysis_result",
+                occurred_at=_utc_now().isoformat(),
+                install_id=config.install_id,
+                payload={
+                    "outcome_status": validated_outcome_status,
+                    "classifiable": outcome_status in {"matched", "unclassified"},
```
[the remark this answers is not in the document]
Comment (tborys) [reply driftshield/src/driftshield/telemetry/service.py]: This is already addressed on `main`. `record_analysis_event()` now derives `classifiable` from the normalized `validated_outcome_status`, so accepted values like `" matched "` stay internally consistent after trimming. Resolving as fixed.
